src/battle_util.c

```diff
@@ -9915,7 +9924,7 @@ static inline u32 CalcDefenseStat(struct DamageCalculationData *damageCalcData,
     }
 
     // Self-destruct / Explosion cut defense in half
-    if (B_EXPLOSION_DEFENSE < GEN_5 && gMovesInfo[gCurrentMove].effect == EFFECT_EXPLOSION)
+    if (B_EXPLOSION_DEFENSE < GEN_5 && GetMoveEffect(gCurrentMove) == EFFECT_EXPLOSION)
```

Comment:
I believe you forgot to use the moveEffect you defined at the top here

Reply:
Fixed